magicctapipe/scripts/lst1_magic/semi_automatic_scripts/coincident_events.py

```diff
@@ -0,0 +1,275 @@
+    rc_lines,
+    slurm_lines,
+)
+
+__all__ = ["configfile_coincidence", "linking_bash_lst"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def configfile_coincidence(ids, target_dir, source_name, NSB_match):
+
+    """
+    This function creates the configuration file needed for the event coincidence step
+
+    Parameters
+    ----------
+    ids : list
+        List of telescope IDs
+    target_dir : str
+        Path to the working directory
+    source_name : str
+        Name of the target source
+    NSB_match : bool
+        If real data are matched to pre-processed MCs or not
+    """
+
+    config_file = resource_file("config.yaml")
+    with open(
+        config_file, "rb"
+    ) as fc:  # "rb" mode opens the file in binary format for reading
+        config_dict = yaml.safe_load(fc)
+    coincidence = config_dict["event_coincidence"]
+
+    conf = {}
+    conf["event_coincidence"] = coincidence
+
+    if not NSB_match:
+        file_name = f"{target_dir}/v{__version__}/{source_name}/config_coincidence.yaml"
+    else:
+        file_name = f"{target_dir}/v{__version__}/{source_name}/config_coincidence.yaml"
+    with open(file_name, "w") as f:
+        lines = [
+            "mc_tel_ids:",
+            f"\n    LST-1: {ids[0]}",
+            f"\n    LST-2: {ids[1]}",
+            f"\n    LST-3: {ids[2]}",
+            f"\n    LST-4: {ids[3]}",
+            f"\n    MAGIC-I: {ids[4]}",
+            f"\n    MAGIC-II: {ids[5]}",
+            "\n",
+        ]
+        f.writelines(lines)
+        yaml.dump(conf, f, default_flow_style=False)
+
+
+def linking_bash_lst(
+    target_dir, LST_runs, source_name, LST_version, env_name, NSB_match, cluster
+):
+
+    """
+    This function links the LST data paths to the working directory and creates bash scripts.
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    LST_runs : matrix of strings
+        This matrix ([['date','run'],['date','run']...]) is imported from *_LST_runs.txt files and tells the function where to find the LST data and link them to our working directory
+    source_name : str
+        Target name
+    LST_version : str
+        The lstchain version used to process the LST data
+    env_name : str
+        Name of the conda environment
+    NSB_match : bool
+        If real data are matched to pre-processed MCs or not
+    """
+
+    if NSB_match:
+        coincidence_DL1_dir = f"{target_dir}/v{__version__}/{source_name}"
+
+        MAGIC_DL1_dir = f"{target_dir}/v{__version__}/{source_name}/DL1"
+    else:
+        coincidence_DL1_dir = (
+            f"{target_dir}/v{__version__}/{source_name}/DL1/Observations"
+        )
+        MAGIC_DL1_dir = f"{target_dir}/v{__version__}/{source_name}/DL1/Observations/"
+
+    dates = [os.path.basename(x) for x in glob.glob(f"{MAGIC_DL1_dir}/Merged/Merged_*")]
+    if cluster == 'SLURM':
```

Comment:
for the moment 'SLURM' is the only option, but when we introduce other clusters a lot of code can be shared between them, so we should not have if cluster == AAA < long code> elif cluster == BBB <similar long code>.
but rather set proper variables (like paths) before the bulk of the code. 
Since it is difficult to know now which variables will need to be set how about you make a simple if cluster != 'SLURM' <exit with an error message> and have the rest of the code one indent less complicated :-)

Reply:
Done